### PR TITLE
feat(wallet): Generate New Addresses

### PR DESCRIPTION
**Pull request by coreyphillips**
This PR:
- Adds and implements the `generateNewReceiveAddress` method when creating Bitcoin invoices.
- Adds `lastUsedAddressIndex` & `lastUsedChangeAddressIndex` to wallet reducer to track the gap limit as new addresses are generated.
  - The gap limit is only enforced when attempting to generate new addresses and is not yet implemented when scanning addresses for balances.

### CODE REVIEW

File: src/screens/Wallets/Receive/index.tsx [hunk 1]
```diff
@@ -40,76 +48,109 @@ const Receive = ({ navigation }): ReactElement => {
...
-	}, [amount, message]);
+	useEffect(() => {
+		resetInvoice();
+	}, []);
+
+	useEffect(() => {
+		setInvoiceDetails().then();
+		getLightningInvoice().then();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [amount, message, selectedNetwork, selectedWallet]);
```
Comment (pwltr) [inline src/screens/Wallets/Receive/index.tsx]: `getLightningInvoice()` is called twice here on mount, once in `setInvoiceDetails()`. Also the `.then()` doesn't do anything here and should be omitted.

File: src/utils/wallet/index.ts [hunk 2]
```diff
@@ -2539,3 +2540,56 @@ export const getBalance = ({
...
+			selectedWallet = getSelectedWallet();
+		}
+		if (!addressType) {
+			addressType = getSelectedAddressType({ selectedNetwork, selectedWallet });
+		}
+		const { currentWallet } = getCurrentWallet({
+			selectedWallet,
+			selectedNetwork,
+		});
+		const addressIndex = currentWallet.addressIndex[addressType].index;
```
Comment (pwltr) [inline src/utils/wallet/index.ts]: This throws an error for me, leading to no new address being generated: `currentWallet.addressIndex[addressType]` is undefined.

**Review by pwltr — CHANGES_REQUESTED** [review]

File: src/screens/Wallets/Receive/index.tsx (hunk 1 above)
Comment (coreyphillips) [reply src/screens/Wallets/Receive/index.tsx]: Fixed [here](https://github.com/synonymdev/bitkit/pull/286/commits/d0471e287bf139c0d9502c90790aa4f835fa6234).

The `.then()` is to acknowledge that a Promise is being returned and to remove the warning from certain IDE's.
Screen Shot 2022-08-29 at 11 59 39 AM

File: src/utils/wallet/index.ts (hunk 2 above)
Comment (coreyphillips) [reply src/utils/wallet/index.ts]: Nice. Fixed [here](https://github.com/synonymdev/bitkit/pull/286/commits/5d4fa5fafcd1f82689e8696f82a11466ae8a1a48).

File: src/screens/Wallets/Receive/index.tsx (hunk 1 above)
Comment (pwltr) [reply src/screens/Wallets/Receive/index.tsx]: Interesting, never seen this but seems useful. Shame that ESLint doesn't have a rule for this.

**Review by pwltr — APPROVED** [review]

**Pull request merged**